authorization/invitation/repository/invitation_blackbox_test.go

```diff
@@ -0,0 +1,131 @@
+package repository_test
+
+import (
+	"testing"
+
+	invitationRepo "github.com/fabric8-services/fabric8-auth/authorization/invitation/repository"
+	roleRepo "github.com/fabric8-services/fabric8-auth/authorization/role/repository"
+	"github.com/fabric8-services/fabric8-auth/gormtestsupport"
+	testsupport "github.com/fabric8-services/fabric8-auth/test"
+
+	"github.com/fabric8-services/fabric8-auth/authorization"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type invitationBlackBoxTest struct {
+	gormtestsupport.DBTestSuite
+	repo invitationRepo.InvitationRepository
+}
+
+func TestRunInvitationBlackBoxTest(t *testing.T) {
+	suite.Run(t, &invitationBlackBoxTest{DBTestSuite: gormtestsupport.NewDBTestSuite()})
+}
+
+func (s *invitationBlackBoxTest) SetupTest() {
+	s.DBTestSuite.SetupTest()
+	s.DB.LogMode(true)
+	s.repo = invitationRepo.NewInvitationRepository(s.DB)
+}
+
+func (s *invitationBlackBoxTest) TestOKToDelete() {
+	invitation, err := s.CreateTestInvitation()
+	require.NoError(s.T(), err)
+
+	invitations, err := s.repo.ListForIdentity(s.Ctx, *invitation.InviteTo)
+	require.NoError(s.T(), err)
+	require.Equal(s.T(), 1, len(invitations))
+
+	err = s.repo.Delete(s.Ctx, invitation.InvitationID)
+	require.NoError(s.T(), err)
+
+	invitations, err = s.repo.ListForIdentity(s.Ctx, *invitation.InviteTo)
+	require.NoError(s.T(), err)
+	require.Equal(s.T(), 0, len(invitations))
+}
+
+func (s *invitationBlackBoxTest) TestOKToLoad() {
+	invitation, err := s.CreateTestInvitation()
+	require.NoError(s.T(), err)
+
+	_, err = s.repo.Load(s.Ctx, invitation.InvitationID)
+	require.NoError(s.T(), err)
+}
+
+func (s *invitationBlackBoxTest) TestExistsInvitation() {
+	invitation, err := s.CreateTestInvitation()
+	require.NoError(s.T(), err)
+
+	exists, err := s.repo.CheckExists(s.Ctx, invitation.InvitationID)
+	require.NoError(s.T(), err)
+	require.True(s.T(), exists)
+}
+
+func (s *invitationBlackBoxTest) TestOKToSave() {
+	invitation, err := s.CreateTestInvitation()
+	require.NoError(s.T(), err)
+
+	_, err = s.repo.Load(s.Ctx, invitation.InvitationID)
+	require.NoError(s.T(), err)
+
+	require.False(s.T(), invitation.Member)
+
+	invitation.Member = true
+	err = s.repo.Save(s.Ctx, &invitation)
+	require.NoError(s.T(), err)
+
+	require.True(s.T(), invitation.Member)
```

Comment:
Check all the relevant field of the saved invitation.

Reply:
Done